serverless-plugin-canary-deployments.js

```diff
@@ -9,9 +9,13 @@ class ServerlessCanaryDeployments {
     this.awsProvider = this.serverless.getProvider('aws');
     this.naming = this.awsProvider.naming;
     this.service = this.serverless.service;
-    this.hooks = {
-      'before:package:finalize': this.addCanaryDeploymentResources.bind(this)
-    };
+    this.preProdDeployment = _.prop(['custom', 'deploymentSettings', 'preProdDeployment'], this.service)
+
+    if(!this.preProdDeployment) {
```

Comment:
Is this where you's expect this to be?

Reply:
I was intending on leaving this undocumented until you figure out the best way, for us it's a switch that mean dev, qa don't have canary deployments.

Reply:
In Serverless different environments are managed through stages, so I think it would make more sense to have a property `excludeStages` or something like that:

```yaml
custom:
  deploymentSettings:
    excludeStages:
      - pre
      - prod
```
Then, the check should probably take place in the `addCanaryDeploymentResources` method. We're already checking if there's any function with deployment settings and skipping all the CF resources generation if there's none. A good approach would be extracting this check into another method and including also the `excludeStages` check.